File: Programmers/Lv.3/여행 경로.js
function solution(tickets) {
  tickets.sort();
  let answer = [];
  const ch = Array.from({ length: tickets.length }, () => 0);
  let isStop = false;
  const DFS = (start, tmp) => {
    if (isStop) return;
    if (tmp.length === tickets.length) {
      tmp.push(start);
      isStop = true;
      answer.push(tmp);
    } else {
      for (let i = 0; i < tickets.length; i++) {
        if (start === tickets[i][0] && ch[i] === 0) {
          ch[i] = 1;
          const tm = [...tmp];
          tm.push(start);
          DFS(tickets[i][1], tm);
          ch[i] = 0;
        }
      }
    }
  };
  DFS("ICN", []);

  return answer.shift();
}

File: python/Programmers/Lv.3/여행경로.py
def solution(tickets):
    routes = {}
    for t in tickets:
        routes[t[0]] = routes.get(t[0],[]) + [t[1]]
    for r in routes:
        routes[r].sort(reversed=True)
    stack = ['ICN']
    path = []
    while len(stack) > 0 :
        top = stack[-1]
        if top not in routes or len(routes[top]) == 0:
            path.append(routes.pop())
        else:
            stack.append(routes[top][-1])
            routes[top] = routes[top][:-1]
    return path[::-1]
